lib/screens/settings/quick_links.dart

```diff
@@ -75,67 +69,81 @@ class _QuickLinksScreenState extends State<QuickLinksScreen> {
         Padding(
           padding: const EdgeInsets.only(bottom: 16),
           child: Form(
-            key: _formKey,
-            child: Row(children: [
-              PopupMenuButton<String>(
-                icon: Icon(quickLinksIconsMap[_activeIcon] ?? Icons.link),
-                onSelected: (result) {
-                  setState(() {
-                    _activeIcon = result;
-                  });
-                },
-                itemBuilder: (context) {
-                  final entries = quickLinksIconsMap.entries.toList()
-                    ..sort((a, b) => a.key.compareTo(b.key));
-                  return entries
-                      .map((entry) => PopupMenuItem(
-                          value: entry.key, child: Icon(entry.value)))
-                      .toList();
-                },
-              ),
-              Expanded(
-                child: TextFormField(
-                    controller: _textEditingController,
-                    decoration: InputDecoration(
-                        hintText: "URL",
-                        suffixIcon: IconButton(
-                            icon: const Icon(Icons.qr_code_scanner),
-                            onPressed: () {
-                              showModalBottomSheet<void>(
-                                  context: context,
-                                  builder: (context) {
-                                    return MobileScanner(
-                                        allowDuplicates: false,
-                                        onDetect: (barcode, args) {
-                                          final code = barcode.rawValue;
-                                          if (code != null) {
-                                            _textEditingController.text = code;
-                                          }
-                                          Navigator.of(context).pop();
-                                        });
-                                  });
-                            })),
-                    validator: (value) {
-                      if (value == null ||
-                          value.isEmpty ||
-                          Uri.tryParse(value) == null) {
-                        return "This doesn't look like a valid URL.";
-                      }
-                      final quickLinksModel =
-                          Provider.of<QuickLinksModel>(context, listen: false);
-                      if (quickLinksModel.sources.any(
-                          (s) => s.url.toString() == Uri.encodeFull(value))) {
-                        return "This link already exists";
-                      }
-                      return null;
+              key: _formKey,
+              child: Column(children: [
+                TextFormField(
+                  controller: _labelEditingController,
+                  decoration: const InputDecoration(
+                    labelText: "Label",
```

Comment:
I don't think we need to label it if we have the hint, it's a bit duplicitous:

![image](https://user-images.githubusercontent.com/511342/169924394-60a6b6b4-f6c5-49de-8bb5-9e80a919fe77.png)

Reply:
Actually, the hint should be the auto-fetched metadata name if possible. For example, if I enter google.com as the URL the label hint should be "Google"